src/prefect/workers/base.py

```diff
@@ -1252,7 +1254,12 @@ async def _mark_flow_run_as_cancelled(
         state_updates.setdefault("type", StateType.CANCELLED)
         if TYPE_CHECKING:
             assert flow_run.state
-        state = flow_run.state.model_copy(update=state_updates)
+
+        if flow_run.state:
+            state = flow_run.state.model_copy(update=state_updates)
+        else:
+            # Unexpectedly when flow run does not have a state, create a new one
+            state: "State[R]" = Cancelled()
```

Comment:
I left the `state_updates` out of my suggestion before, but we probably want them here to ensure the message is added to the state. Also, I don't think the type hint is necessary, and the `R` TypeVar in this module wouldn't be the correct type for the `State` generic here.
```suggestion
            state = Cancelled(**state_updates)
```